### DISCUSSION TITLE
Pre-packaged OMZ+plugins for non-git outgoing environments

### DISCUSSION THREAD

**Discussion by hvisage** [ohmyzsh/ohmyzsh#10204] [Help] [answered]
I have a need to pre-package a set of OMZ+plugins for Windows/GUI users that'll be using it on instances without direct/git access to the internet (ie. only incoming SSH)

My first attempt to "just" clone/copy the .zsh/.antigen/oh-my-zsh directories, I got hit with loads of user specific directories that needed to be manually fixed up.

I noticed Debian's .deb of antigen, and looking for something similar for OMZ and the needed plugins, but even if I can just pre-package and then deploy/extra for the users (will be done with something Ansible) will be a great solution as I can then create the needed and just deploy everywhere and have a uniform shell experience for these users.

Thus: Is there OMZ+plugins packages I've missed? (Specifically for Debian and Ubuntu for now)
IF no packages, how best to package for the users?

Any other advice for this case where the deployed user's instances doesn't have internet outgoing access?

**Comment by mcornella** [comment, marked as answer]
You can 

1. download the zip file ([link to current master](https://github.com/ohmyzsh/ohmyzsh/archive/refs/heads/master.zip)) and extract it wherever you need, and 
2. copy the `templates/zshrc.zsh-template` file to `~/.zshrc` and modify the `export ZSH` line to point to where you extracted the zip file, as well as disable the update logic by setting `DISABLE_AUTO_UPDATE=true` in the zshrc file, before the `source $ZSH...` line.

If you want to skip (2), you can just add these lines to an existing zshrc file:
```zsh
export ZSH=<path-to-extracted-zip>
DISABLE_AUTO_UPDATE=true
plugins=(<whatever-you-want>)
source $ZSH/oh-my-zsh.sh
```

Ideally these lines would go at the top of the zshrc file (though **after any `$PATH` changes**), so that any later modifications take effect and aren't overwritten by OMZ.
